src/components/lib/sl/sl.c

```diff
@@ -595,9 +596,10 @@ sl_init(microsec_t period)
 }
 
 void
-sl_sched_loop(void)
+sl_sched_loop(int non_block)
 {
-	struct sl_global *g = sl__globals();
+	struct sl_global *g   = sl__globals();
+	rcv_flags_t       rfl = (non_block ? RCV_NON_BLOCKING : 0) | RCV_ALL_PENDING;
```

Comment:
OK, so you're just using `non_block` as binary, but then are assigning a name to it.  This seems strange.

I can't imagine this should be an argument at the API level.  `sl_sched_loop()` should remain for default use, and you can create a `sl_sched_loop_nonblock()` for whatever this is all for.

You keep wanting to make the API more complex for common usage.  You need to push back against this impulse.

All of that said, I have no idea why want this to be nonblocking.

Reply:
Yes, I had the same feeling that I made the API a bit confusing to use. 
I have mentioned the reasoning for nonblocking API briefly in the commit message, but I will hold this PR for now and discuss about it the next time we meet.